core/trino-main/src/test/java/io/trino/type/TestDecimalOperators.java

```diff
@@ -194,14 +214,17 @@ public void testDivide()
         assertDecimalFunction("DECIMAL '100.00' / DECIMAL '0.3'", decimal("0333.33"));
         assertDecimalFunction("DECIMAL '100.00' / DECIMAL '0.30'", decimal("00333.33"));
         assertDecimalFunction("DECIMAL '100.00' / DECIMAL '-0.30'", decimal("-00333.33"));
+        assertDecimalFunction("DECIMAL '-100.00' / DECIMAL '0.30'", decimal("-00333.33"));
         assertDecimalFunction("DECIMAL '200.00' / DECIMAL '0.3'", decimal("0666.67"));
         assertDecimalFunction("DECIMAL '200.00000' / DECIMAL '0.3'", decimal("0666.66667"));
         assertDecimalFunction("DECIMAL '200.00000' / DECIMAL '-0.3'", decimal("-0666.66667"));
+        assertDecimalFunction("DECIMAL '-200.00000' / DECIMAL '0.3'", decimal("-0666.66667"));
```

Comment:
Should you add `-0 / DECIMAL xx` case too?

Reply:
I think there is one in line 227?

Reply:
Oh, I think I misunderstood. I don't think it's possible to create a negative zero using the `DECIMAL` literal, but I'll have to check.